tests/integration_tests/test_command_line_options.py

```diff
@@ -103,10 +113,14 @@ def setUp(self):
         self.data_file = "data.yaml"
         with open(self.data_file, "w") as f:
             f.write("hello: world")
-        self.patcher = patch(
+        self.patcher1 = patch(
```

Comment:
this patcher1 won't work because the location of the function has been changed

Reply:
Although the function is defined in engine_factory.py, it is being imported into engine.py, hence requiring a patcher. Without this patcher, the tests will fail - I have verified that both patchers are necessary.